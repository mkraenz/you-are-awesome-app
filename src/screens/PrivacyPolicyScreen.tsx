import { useNavigation } from "@react-navigation/native";
import React from "react";
import { useTranslation } from "react-i18next";
import { Linking, StyleSheet } from "react-native";
import { Button } from "react-native-paper";
import { WebView } from "react-native-webview";
import Layout from "../components/common/Layout";
import { Route } from "../navigation/Route";

const styles = StyleSheet.create({
    viewInBrowserButton: {
        marginTop: 16,
        marginHorizontal: 16,
        marginBottom: 8,
    },
});

const PrivacyPolicyScreen: React.FC = () => {
    const { t } = useTranslation();
    const navigation = useNavigation();
    return (
        <Layout
            route={Route.PrivacyPolicy}
            onBack={() => navigation.goBack()}
            title={t(Route.PrivacyPolicy)}
        >
            <Button
                mode="outlined"
                style={styles.viewInBrowserButton}
                onPress={() => Linking.openURL(t("privacyPolicyLink"))}
                accessibilityStates={{}}
            >
                {t("viewInBrowser")}
            </Button>
            <WebView source={{ uri: t("privacyPolicyLink") }} />
        </Layout>
    );
};

export default PrivacyPolicyScreen;
