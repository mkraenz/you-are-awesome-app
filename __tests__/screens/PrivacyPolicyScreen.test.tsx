import React, { FC } from "react";
import "react-native";
import { DefaultTheme, Provider as PaperProvider } from "react-native-paper";
// Note: test renderer must be required after react-native.
import renderer from "react-test-renderer";
import PrivacyPolicyScreen from "../../src/screens/PrivacyPolicyScreen";
import MockedNavigator from "../helpers/MockedNavigation";
import TestLocalizationProvider from "../helpers/TestLocalizationProvider";

jest.mock("react-native-webview", () => ({ WebView: "WebView" }));
// Silence the warning https://github.com/facebook/react-native/issues/11094#issuecomment-263240420
jest.mock("react-native/Libraries/Animated/src/NativeAnimatedHelper");

const ConfiguredPrivacyPolicyScreen: FC = () => (
    <PaperProvider theme={DefaultTheme}>
        <TestLocalizationProvider>
            <MockedNavigator component={PrivacyPolicyScreen} />
        </TestLocalizationProvider>
    </PaperProvider>
);

it("renders a button and the webview", async () => {
    const tree = renderer.create(<ConfiguredPrivacyPolicyScreen />).toJSON();

    expect(tree).toMatchSnapshot();
});
