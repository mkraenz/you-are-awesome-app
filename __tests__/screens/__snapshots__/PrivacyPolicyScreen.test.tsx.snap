// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a button and the webview 1`] = `
<View
  collapsable={false}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <RNCSafeAreaProvider
      onInsetsChange={[Function]}
      style={
        Array [
          Object {
            "flex": 1,
          },
          undefined,
        ]
      }
    >
      <View
        pointerEvents="box-none"
        style={
          Array [
            Object {
              "zIndex": 1,
            },
            false,
          ]
        }
      >
        <View
          accessibilityElementsHidden={false}
          importantForAccessibility="auto"
          onLayout={[Function]}
          pointerEvents="box-none"
          style={null}
        >
          <View
            pointerEvents="box-none"
            style={
              Object {
                "bottom": 0,
                "left": 0,
                "opacity": 1,
                "position": "absolute",
                "right": 0,
                "top": 0,
                "zIndex": 0,
              }
            }
          >
            <View
              style={
                Object {
                  "backgroundColor": "rgb(255, 255, 255)",
                  "borderBottomColor": "rgb(216, 216, 216)",
                  "flex": 1,
                  "shadowColor": "rgb(216, 216, 216)",
                  "shadowOffset": Object {
                    "height": 0.5,
                    "width": 0,
                  },
                  "shadowOpacity": 0.85,
                  "shadowRadius": 0,
                }
              }
            />
          </View>
          <View
            pointerEvents="box-none"
            style={
              Object {
                "height": 64,
                "maxHeight": undefined,
                "minHeight": undefined,
                "opacity": undefined,
                "transform": undefined,
              }
            }
          >
            <View
              pointerEvents="none"
              style={
                Object {
                  "height": 20,
                }
              }
            />
            <View
              pointerEvents="box-none"
              style={
                Object {
                  "alignItems": "center",
                  "flex": 1,
                  "flexDirection": "row",
                  "justifyContent": "center",
                }
              }
            >
              <View
                pointerEvents="box-none"
                style={
                  Object {
                    "marginHorizontal": 16,
                    "opacity": 1,
                  }
                }
              >
                <Text
                  accessibilityRole="header"
                  numberOfLines={1}
                  onLayout={[Function]}
                  style={
                    Object {
                      "color": "rgb(28, 28, 30)",
                      "fontSize": 17,
                      "fontWeight": "600",
                    }
                  }
                >
                  MockedScreen
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={
          Object {
            "flex": 1,
          }
        }
      >
        <View
          pointerEvents="box-none"
          style={
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
        >
          <View
            collapsable={false}
            style={
              Object {
                "opacity": 1,
              }
            }
          />
          <View
            accessibilityElementsHidden={false}
            closing={false}
            gestureVelocityImpact={0.3}
            importantForAccessibility="auto"
            onClose={[Function]}
            onGestureBegin={[Function]}
            onGestureCanceled={[Function]}
            onGestureEnd={[Function]}
            onOpen={[Function]}
            onTransitionStart={[Function]}
            pointerEvents="box-none"
            style={
              Object {
                "bottom": 0,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
              }
            }
            transitionSpec={
              Object {
                "close": Object {
                  "animation": "spring",
                  "config": Object {
                    "damping": 500,
                    "mass": 3,
                    "overshootClamping": true,
                    "restDisplacementThreshold": 10,
                    "restSpeedThreshold": 10,
                    "stiffness": 1000,
                  },
                },
                "open": Object {
                  "animation": "spring",
                  "config": Object {
                    "damping": 500,
                    "mass": 3,
                    "overshootClamping": true,
                    "restDisplacementThreshold": 10,
                    "restSpeedThreshold": 10,
                    "stiffness": 1000,
                  },
                },
              }
            }
          >
            <View
              pointerEvents="box-none"
              style={
                Object {
                  "flex": 1,
                }
              }
            >
              <View
                collapsable={false}
                forwardedRef={[Function]}
                onGestureHandlerEvent={[Function]}
                onGestureHandlerStateChange={[Function]}
                style={
                  Object {
                    "flex": 1,
                    "transform": Array [
                      Object {
                        "translateX": 0,
                      },
                      Object {
                        "translateX": 0,
                      },
                    ],
                  }
                }
              >
                <View
                  pointerEvents="none"
                  style={
                    Object {
                      "backgroundColor": "rgb(242, 242, 242)",
                      "bottom": 0,
                      "left": 0,
                      "position": "absolute",
                      "shadowColor": "#000",
                      "shadowOffset": Object {
                        "height": 1,
                        "width": -1,
                      },
                      "shadowOpacity": 0.3,
                      "shadowRadius": 5,
                      "top": 0,
                      "width": 3,
                    }
                  }
                />
                <View
                  style={
                    Array [
                      Object {
                        "flex": 1,
                        "overflow": "hidden",
                      },
                      Array [
                        Object {
                          "backgroundColor": "rgb(242, 242, 242)",
                        },
                        undefined,
                      ],
                    ]
                  }
                >
                  <View
                    style={
                      Object {
                        "flex": 1,
                        "flexDirection": "column-reverse",
                      }
                    }
                  >
                    <View
                      style={
                        Object {
                          "flex": 1,
                        }
                      }
                    >
                      <View
                        accessibilityStates={Object {}}
                        style={
                          Object {
                            "backgroundColor": "#ffffff",
                            "height": "100%",
                            "shadowColor": "#000000",
                            "shadowOffset": Object {
                              "height": 3,
                              "width": 0,
                            },
                            "shadowOpacity": 0.24,
                            "shadowRadius": 4,
                            "width": "100%",
                          }
                        }
                      >
                        <RCTSafeAreaView
                          emulateUnlessSupported={true}
                          style={
                            Array [
                              Object {
                                "backgroundColor": "#6200ee",
                                "elevation": 4,
                                "zIndex": 0,
                              },
                              Object {
                                "shadowColor": "#000000",
                                "shadowOffset": Object {
                                  "height": 3,
                                  "width": 0,
                                },
                                "shadowOpacity": 0.24,
                                "shadowRadius": 4,
                              },
                            ]
                          }
                        >
                          <View
                            accessibilityStates={Object {}}
                            style={
                              Object {
                                "alignItems": "center",
                                "backgroundColor": "#6200ee",
                                "elevation": 0,
                                "flexDirection": "row",
                                "height": 56,
                                "marginTop": null,
                                "paddingHorizontal": 4,
                              }
                            }
                          >
                            <View
                              accessibilityLabel="Back"
                              accessibilityRole="button"
                              accessible={true}
                              focusable={true}
                              hitSlop={
                                Object {
                                  "bottom": 6,
                                  "left": 6,
                                  "right": 6,
                                  "top": 6,
                                }
                              }
                              onClick={[Function]}
                              onResponderGrant={[Function]}
                              onResponderMove={[Function]}
                              onResponderRelease={[Function]}
                              onResponderTerminate={[Function]}
                              onResponderTerminationRequest={[Function]}
                              onStartShouldSetResponder={[Function]}
                              style={
                                Array [
                                  Object {
                                    "overflow": "hidden",
                                  },
                                  Array [
                                    Object {
                                      "alignItems": "center",
                                      "justifyContent": "center",
                                      "margin": 6,
                                      "overflow": "hidden",
                                    },
                                    Object {
                                      "borderRadius": 18,
                                      "height": 36,
                                      "width": 36,
                                    },
                                    undefined,
                                    undefined,
                                  ],
                                ]
                              }
                            >
                              <View>
                                <View
                                  style={
                                    Array [
                                      Object {
                                        "alignItems": "center",
                                        "justifyContent": "center",
                                      },
                                      Object {
                                        "height": 24,
                                        "width": 24,
                                      },
                                    ]
                                  }
                                >
                                  <View
                                    style={
                                      Object {
                                        "bottom": 0,
                                        "left": 0,
                                        "opacity": 1,
                                        "position": "absolute",
                                        "right": 0,
                                        "top": 0,
                                        "transform": Array [
                                          Object {
                                            "rotate": "0deg",
                                          },
                                        ],
                                      }
                                    }
                                  >
                                    <View
                                      style={
                                        Array [
                                          Object {
                                            "alignItems": "center",
                                            "justifyContent": "center",
                                          },
                                          Object {
                                            "height": 24,
                                            "transform": Array [
                                              Object {
                                                "scaleX": 1,
                                              },
                                            ],
                                            "width": 24,
                                          },
                                        ]
                                      }
                                    >
                                      <Image
                                        source={1}
                                        style={
                                          Array [
                                            Object {
                                              "height": 21,
                                              "resizeMode": "contain",
                                              "width": 21,
                                            },
                                            Object {
                                              "tintColor": "#ffffff",
                                            },
                                          ]
                                        }
                                      />
                                    </View>
                                  </View>
                                </View>
                              </View>
                            </View>
                            <View
                              accessibilityStates={Object {}}
                              accessible={true}
                              focusable={false}
                              onClick={[Function]}
                              onResponderGrant={[Function]}
                              onResponderMove={[Function]}
                              onResponderRelease={[Function]}
                              onResponderTerminate={[Function]}
                              onResponderTerminationRequest={[Function]}
                              onStartShouldSetResponder={[Function]}
                              style={
                                Array [
                                  Object {
                                    "flex": 1,
                                    "paddingHorizontal": 12,
                                  },
                                  Array [
                                    Object {
                                      "marginLeft": 8,
                                    },
                                    Object {
                                      "alignItems": "center",
                                    },
                                    Object {},
                                  ],
                                ]
                              }
                            >
                              <Text
                                accessibilityRole="header"
                                accessibilityTraits="header"
                                accessible={true}
                                numberOfLines={1}
                                style={
                                  Array [
                                    Object {
                                      "color": "#000000",
                                      "fontFamily": "System",
                                      "fontWeight": "400",
                                      "textAlign": "left",
                                    },
                                    Array [
                                      Object {
                                        "color": "#ffffff",
                                        "fontFamily": "System",
                                        "fontWeight": "400",
                                      },
                                      Object {
                                        "fontSize": 17,
                                      },
                                      undefined,
                                    ],
                                  ]
                                }
                              >
                                Privacy Policy
                              </Text>
                            </View>
                          </View>
                        </RCTSafeAreaView>
                        <View
                          style={
                            Object {
                              "flex": 1,
                              "paddingHorizontal": 16,
                            }
                          }
                        >
                          <View
                            accessibilityStates={Object {}}
                            style={
                              Object {
                                "backgroundColor": "#6200ee",
                                "borderColor": "transparent",
                                "borderRadius": 4,
                                "borderStyle": "solid",
                                "borderWidth": 0,
                                "elevation": 2,
                                "margin": 16,
                                "minWidth": 64,
                                "shadowColor": "#000000",
                                "shadowOffset": Object {
                                  "height": 0.75,
                                  "width": 0,
                                },
                                "shadowOpacity": 0.24,
                                "shadowRadius": 1.5,
                              }
                            }
                          >
                            <View
                              accessibilityRole="button"
                              accessible={true}
                              focusable={true}
                              onClick={[Function]}
                              onResponderGrant={[Function]}
                              onResponderMove={[Function]}
                              onResponderRelease={[Function]}
                              onResponderTerminate={[Function]}
                              onResponderTerminationRequest={[Function]}
                              onStartShouldSetResponder={[Function]}
                              style={
                                Array [
                                  Object {
                                    "overflow": "hidden",
                                  },
                                  Object {
                                    "borderRadius": 4,
                                  },
                                ]
                              }
                            >
                              <View
                                style={
                                  Array [
                                    Object {
                                      "alignItems": "center",
                                      "flexDirection": "row",
                                      "justifyContent": "center",
                                    },
                                    undefined,
                                  ]
                                }
                              >
                                <Text
                                  numberOfLines={1}
                                  style={
                                    Array [
                                      Object {
                                        "color": "#000000",
                                        "fontFamily": "System",
                                        "fontWeight": "400",
                                        "textAlign": "left",
                                      },
                                      Array [
                                        Object {
                                          "letterSpacing": 1,
                                          "marginHorizontal": 16,
                                          "marginVertical": 9,
                                          "textAlign": "center",
                                        },
                                        undefined,
                                        Object {
                                          "textTransform": "uppercase",
                                        },
                                        Object {
                                          "color": "#ffffff",
                                          "fontFamily": "System",
                                          "fontWeight": "500",
                                        },
                                        Object {
                                          "fontFamily": "System",
                                          "fontWeight": "500",
                                        },
                                        undefined,
                                      ],
                                    ]
                                  }
                                >
                                  View in Web Browser
                                </Text>
                              </View>
                            </View>
                          </View>
                          <WebView
                            source={
                              Object {
                                "uri": "https://you-are-awesome-app-privacy-policy.s3.eu-central-1.amazonaws.com/PrivacyPolicyEn.html",
                              }
                            }
                          />
                        </View>
                      </View>
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </RNCSafeAreaProvider>
  </View>
</View>
`;
